### ISSUE TITLE
The lang field does not alter key terms in typst

### ISSUE THREAD

**Issue by wjschne** [quarto-dev/quarto-cli#9555]
### Bug description

When the `lang` field is set in metadata, many key terms are not translated in typst as they are in other formats.

### Steps to reproduce

```yaml
---
title: Mi Título
format: typst
lang: es
abstract: Escribo algo.
---
```

In other formats, the abstract is translated to the language set in the `lang` field.

### Expected behavior

The abstract (and many other terms) should be in Spanish. In this case, the Abstract should be "Resumen"

### Actual behavior

Abstract is not translated. It is "Abstract" instead of "Resumen"

### Your environment

RStudio 2024.04.0+735 "Chocolate Cosmos" Release (a00d0e775dbc93e0d79a1bf474e3e8e8de677383, 2024-04-24) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2024.04.0+735 Chrome/120.0.6099.291 Electron/28.2.6 Safari/537.36, Quarto 1.5.31 (C:/Program Files/Quarto/bin/quarto.exe)

### Quarto check output

Quarto 1.5.31
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.13: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.31
      Path: C:\Program Files\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\renee\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2024

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.3
      Path: C:/Users/renee/AppData/Local/Programs/Python/Python311/python.exe
      Jupyter: 5.3.1
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.4.0
      Path: C:/PROGRA~1/R/R-4.4.0
      LibPaths:
        - C:/Users/renee/AppData/Local/R/win-library/4.4
        - C:/Program Files/R/R-4.4.0/library
      knitr: 1.46.3
      rmarkdown: 2.26

[>] Checking Knitr engine render......OK

**Comment by mcanouil** [comment]
Indeed, that is because, it's currently using the Pandoc template and it hardcodes "abstract" in it.

- https://github.com/quarto-dev/quarto-cli/blob/2b353b5e971eb4017eee06ca0362700ecf69ef2b/src/resources/formats/typst/pandoc/quarto/typst-template.typ#L62-L67

**Comment by wjschne** [comment]
Other possible hard codings:
- https://github.com/quarto-dev/quarto-cli/blob/90af993c237133fe25c0e3ae95bd134aeaee53c1/src/resources/filters/crossref/sections.lua#L75
- https://github.com/quarto-dev/quarto-cli/blob/90af993c237133fe25c0e3ae95bd134aeaee53c1/src/resources/filters/layout/pandoc3_figure.lua#L160

**Comment by gordonwoodhull** [comment]
Also theorem title here:

https://github.com/quarto-dev/quarto-cli/blob/221493577f063df8338152181353bf5caaf61c81/src/resources/filters/customnodes/theorem.lua#L103

**Comment by cderv** [comment]
Regarding the abstract title, 

* Pandoc does not yet support `abstract-title` variable for Typst format. This is used currently to set up the translated value using language for HTML and epub
	https://github.com/quarto-dev/quarto-cli/blob/a604268ea7724d8991512f1f7ed89ce7b84f391a/src/command/render/pandoc.ts#L358-L367
	Pandoc would need to support it, and we would need to support setting a default using language file value

* Though, we have another mechanism using Lua to set some Meta
	https://github.com/quarto-dev/quarto-cli/blob/a604268ea7724d8991512f1f7ed89ce7b84f391a/src/resources/filters/modules/authors.lua#L894-L897
	And we patched HTML templates to use this value instead of regular variable 
	https://github.com/quarto-dev/quarto-cli/blob/a604268ea7724d8991512f1f7ed89ce7b84f391a/src/resources/formats/html/templates/title-metadata.html#L71-L78

Some note for future us that `docx` does support `abtract-title` field, but Docx writer will also set a default based on `lang` field 
(https://github.com/jgm/pandoc/blob/0f29d8195eeb526c4aa14d5f56818f905d5badb5/src/Text/Pandoc/Writers/Docx.hs#L920) 

So we could take approach two at least for solving this. 

> Other possible hard coding:

@wjschne thanks for the report, we should indeed used the translated version here. 

> Also theorem title here:

Good catch @gordonwoodhull ! We should really do a pass on translation for Typst.

I think we can fix what we found for now, and do a full check later, or do you want to do a full check now ?

**Comment by cderv** [comment]
FWIW, issue opened upstream 
- https://github.com/jgm/pandoc/issues/9724

It won't solve it here, but at least Pandoc will gain this feature too.

**Comment by gordonwoodhull** [comment]
Implemented in #9829. We can revert the template change when we get a fix for jgm/pandoc#9724.
